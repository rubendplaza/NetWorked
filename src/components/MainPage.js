import React from "react";
import CardList from "./CardList";
import SearchBox from "./SearchBox";
import Scroll from "./Scroll";
import Header from "./Header";
import "./MainPage.css";

class MainPage extends React.Component {
  componentDidMount() {
    this.props.onRequestConnections();
  }

  filterConnections = () => {
    const { searchField } = this.props;
    return this.props.connections.filter((connection) => {
      return (
        connection.name.first
          .toLowerCase()
          .includes(searchField.toLowerCase()) ||
        connection.name.last.toLowerCase().includes(searchField.toLowerCase())
      );
    });
  };

  render() {
    const { onSearchChange, isPending } = this.props;
    return isPending ? (
      <div className="loadingContainer">
        <h1 className="loadingTag">Loading...</h1>
      </div>
    ) : (
      <div className="tc">
        <Header />
        <SearchBox searchChange={onSearchChange} />
        <Scroll>
          <CardList connections={this.filterConnections()} />
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
