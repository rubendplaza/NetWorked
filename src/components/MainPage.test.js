import React from "react";
import MainPage from "./MainPage";
import { shallow } from "enzyme";

let wrapper;
beforeEach(() => {
  const mockProps = {
    onRequestConnections: jest.fn(),
    connections: [],
    searchField: "",
    isPending: false,
  };
  wrapper = shallow(<MainPage {...mockProps} />);
});
describe("tests for MainPage component", () => {
  test("renders MainPage without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("renders MainPage loading screen without crashing", () => {
    const mockProps1 = {
      onRequestConnections: jest.fn(),
      connections: [
        {
          index: 1,
          name: {
            first: "John",
            last: "Plaza",
          },
          email: "ruben@gmail.com",
          phone: 1234567890,
          company: "Google",
        },
      ],
      searchField: "",
      isPending: true,
    };

    const wrapper1 = shallow(<MainPage {...mockProps1} />);
    expect(wrapper1).toMatchSnapshot();
  });

  test("filters robots correctly", () => {
    const mockProps2 = {
      onRequestConnections: jest.fn(),
      connections: [
        {
          index: 1,
          name: {
            first: "John",
            last: "Plaza",
          },
          email: "ruben@gmail.com",
          phone: 1234567890,
          company: "Google",
        },
        {
          index: 2,
          name: {
            first: "Amy",
            last: "Plaza",
          },
          email: "rube@gmail.com",
          phone: 1234567899,
          company: "Facebook",
        },
      ],
      searchField: "amy",
      isPending: false,
    };
    expect(wrapper.instance().filterConnections()).toEqual([]);

    const wrapper2 = shallow(<MainPage {...mockProps2} />);
    expect(wrapper2.instance().filterConnections()[0].name.first).toEqual(
      "Amy"
    );
  });

  test("filters robots correctly 2", () => {
    const mockProps3 = {
      onRequestConnections: jest.fn(),
      connections: [
        {
          index: 1,
          name: {
            first: "John",
            last: "Plaza",
          },
          email: "ruben@gmail.com",
          phone: 1234567890,
          company: "Google",
        },
      ],
      searchField: "amy",
      isPending: false,
    };

    const wrapper3 = shallow(<MainPage {...mockProps3} />);
    expect(wrapper3.instance().filterConnections()).toEqual([]);
  });
});
