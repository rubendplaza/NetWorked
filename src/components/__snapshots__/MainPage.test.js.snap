// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for MainPage component renders MainPage loading screen without crashing 1`] = `
<div
  className="loadingContainer"
>
  <h1
    className="loadingTag"
  >
    Loading...
  </h1>
</div>
`;

exports[`tests for MainPage component renders MainPage without crashing 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <CardList
      connections={Array []}
    />
  </Scroll>
</div>
`;
